src/server/game/Handlers/MovementHandler.cpp

```diff
@@ -429,7 +454,65 @@ void WorldSession::HandleMovementOpcodes(WorldPacket & recvData)
         }
     }
 
+    bool jumpopcode = false;
+    if (opcode == MSG_MOVE_JUMP)
+    {
+        jumpopcode = true;
+        if (plrMover && mover->IsFalling())
+        {
+            sLog->outString("MovementHandler::DOUBLE_JUMP by Account id : %u, Player %s", plrMover->GetSession()->GetAccountId(), plrMover->GetName().c_str());
+            sWorld->SendGMText(LANG_GM_ANNOUNCE_DOUBLE_JUMP, plrMover->GetName().c_str());
+            if (sWorld->getBoolConfig(CONFIG_ANTICHEAT_DOUBLEJUMP_ENABLED))
+            {
+                plrMover->GetSession()->KickPlayer();
+                return;
+            }
+        }
+    }
+
+    if (sWorld->getBoolConfig(CONFIG_ANTICHEAT_FAKEJUMPER_ENABLED) && plrMover && !plrMover->UnderACKmount() && mover->IsFalling() && movementInfo.pos.GetPositionZ() > mover->GetPositionZ())
+    {
+        if (!plrMover->IsJumpingbyOpcode())
+        {
+            plrMover->SetJumpingbyOpcode(true);
+            plrMover->SetUnderACKmount();
+        }
+        else if (!movementInfo.flags & MOVEMENTFLAG_ONTRANSPORT)
```

Comment:
Codacy tip: `Boolean result is used in bitwise operation. Clarify expression with parentheses.`

```cpp
else if (!(movementInfo.flags & MOVEMENTFLAG_ONTRANSPORT))
```